@iamanaws I have the same problem building the x86_64-darwin version. For me it seems to fail because `env` complains about a broken interpreter. This might be a rosetta problem, so I'm not sure to count this against this PR. Do you have the same issue?

Extract from the build log:

```log
[134/2773] CC obj/third_party/nasm/nasm/insnsd.o
[135/2773] CC obj/third_party/nasm/nasm/insnsa.o
[136/2773] LINK ./nasm
FAILED: [code=126] nasm 
TOOL_VERSION=1771355546 ../../build/toolchain/apple/linker_driver.py -Wcrl,driver,../../../../../../../store/fv50h2fzmapcpq7qx0rhidg4q6pwi1bf-signal-webrtc-llvm-cc-and-bintools/bin/clang++ -Wcrl,strippath,../../../../../../../store/fv50h2fzmapcpq7qx0rhidg4q6pwi1bf-signal-webrtc-llvm-cc-and-bintools/bin/llvm-strip -Wcrl,installnametoolpath,../../../../../../../store/fv50h2fzmapcpq7qx0rhidg4q6pwi1bf-signal-webrtc-llvm-cc-and-bintools/bin/llvm-install-name-tool  -fuse-ld=lld -Wl,-fatal_warnings -Wl,--icf=all -Wl,--color-diagnostics -Wl,--strict-auto-link -Wl,--no-call-graph-profile-sort --target=x86_64-apple-darwin -Wl,-fixup_chains -no-canonical-prefixes -Wl,-dead_strip -isysroot ../../../../../../../store/0b11sl1s441yhz6fxa6h8nr8h05c8w7p-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk -mmacos-version-min=12.0 -Wl,-ObjC -Wl,--no-warn-duplicate-rpath -Wl,-no_warn_duplicate_libraries -o "./nasm" "@./nasm.rsp" 
sh: ../../build/toolchain/apple/linker_driver.py: /usr/bin/env: bad interpreter: Operation not permitted
[137/2773] CC obj/third_party/libjpeg_turbo/libjpeg12/jccolor.o
[138/2773] CC obj/third_party/libjpeg_turbo/libjpeg12/jcmainct.o
[139/2773] CC obj/third_party/libjpeg_turbo/libjpeg12/jclossls.o
[140/2773] CXX obj/rtc_base/net_helpers/net_helpers.o
[141/2773] CC obj/third_party/libjpeg_turbo/libjpeg12/jcprepct.o
[142/2773] CXX obj/api/units/data_rate/data_rate.o
[143/2773] CC obj/third_party/libjpeg_turbo/libjpeg12/jdapistd.o
[144/2773] CC obj/third_party/libjpeg_turbo/libjpeg12/jcsample.o
[145/2773] CXX obj/api/video/video_rtp_headers/video_content_type.o
[146/2773] CXX obj/api/video/video_rtp_headers/color_space.o
[147/2773] CXX obj/api/rtp_headers/rtp_headers.o
[148/2773] CXX obj/api/video/video_rtp_headers/video_timing.o
[149/2773] CXX obj/rtc_base/logging/logging.o
ninja: build stopped: subcommand failed.
```

--- a/pkgs/by-name/si/signal-desktop/package.nix
+++ b/pkgs/by-name/si/signal-desktop/package.nix
@@ -293,6 +293,7 @@ stdenv.mkDerivation (finalAttrs: {
       ++ lib.optional withAppleEmojis unfree;
     maintainers = with lib.maintainers; [
       eclairevoyant
+      iamanaws
       marcin-serwin
       teutat3s
     ];

--- a/pkgs/by-name/si/signal-desktop/signal-sqlcipher.nix
+++ b/pkgs/by-name/si/signal-desktop/signal-sqlcipher.nix
@@ -10,6 +10,8 @@
   cargo,
   dump_syms,
   python3,
+  xcodebuild,
+  cctools,
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "node-sqlcipher";
@@ -45,6 +47,10 @@ stdenv.mkDerivation (finalAttrs: {
     cargo
     dump_syms
     python3
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    xcodebuild
+    cctools.libtool
   ];
 
   buildPhase = ''
@@ -75,6 +81,6 @@ stdenv.mkDerivation (finalAttrs: {
       # deps/sqlcipher
       bsd3
     ];
-    platforms = lib.platforms.linux;
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
   };
 })

--- a/pkgs/by-name/si/signal-desktop/webrtc.nix
+++ b/pkgs/by-name/si/signal-desktop/webrtc.nix
@@ -4,6 +4,7 @@
   buildPackages,
   ninja,
   gn,
+  symlinkJoin,
   python3,
   pkg-config,
   glib,
@@ -12,9 +13,20 @@
   writeShellScriptBin,
   gclient2nix,
   rustc,
+  apple-sdk,
+  xcodebuild,
+  llvmPackages,
 }:
 
 let
+  llvmCcAndBintools = symlinkJoin {
+    name = "signal-webrtc-llvm-cc-and-bintools";
+    paths = [
+      llvmPackages.llvm
+      llvmPackages.stdenv.cc
+    ];
+  };
+
   chromiumRosettaStone = {
     cpu =
       platform:
@@ -39,6 +51,13 @@ stdenv.mkDerivation (finalAttrs: {
   gclientDeps = gclient2nix.importGclientDeps ./webrtc-sources.json;
   sourceRoot = "src";
 
+  # Chromium's Darwin toolchain defines _LIBCPP_HARDENING_MODE itself; keep
+  # cc-wrapper from injecting a conflicting default.
+  hardeningDisable = lib.optionals stdenv.isDarwin [
+    "libcxxhardeningfast"
+    "libcxxhardeningextensive"
+  ];
+
   nativeBuildInputs = [
     gn
     ninja
@@ -49,21 +68,44 @@ stdenv.mkDerivation (finalAttrs: {
     rustc
     pkg-config
     gclient2nix.gclientUnpackHook
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    apple-sdk
+    xcodebuild
   ];
 
   buildInputs = [
     glib
-    alsa-lib
     pulseaudio
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    llvmPackages.compiler-rt
+  ]
+  ++ lib.optionals stdenv.isLinux [
+    alsa-lib
   ];
 
   postPatch = ''
     substituteInPlace build/toolchain/linux/BUILD.gn \
       --replace-fail 'toolprefix = "aarch64-linux-gnu-"' 'toolprefix = ""'
-    patchShebangs build/mac/should_use_hermetic_xcode.py
 
     substituteInPlace modules/audio_device/linux/pulseaudiosymboltable_linux.cc \
       --replace-fail "libpulse.so.0" "${pulseaudio}/lib/libpulse.so.0"
+  ''
+  + lib.optionalString stdenv.isDarwin ''
+    # Fix Darwin Python script shebangs for sandbox builds
+    patchShebangs build/mac/should_use_hermetic_xcode.py build/toolchain/apple/linker_driver.py
+
+    substituteInPlace build/config/clang/BUILD.gn \
+      --replace-fail '_clang_lib_dir = "$clang_base_path/lib/clang/$clang_version/lib"' \
+                     '_clang_lib_dir = "${llvmPackages.compiler-rt}/lib"'
+
+    # Keep target triple aligned with nix cc-wrapper expectations to avoid
+    # arm64-apple-macos vs arm64-apple-darwin warning spam.
+    substituteInPlace build/config/mac/BUILD.gn \
+      --replace-fail "apple-macos" "apple-darwin"
+  ''
+  + lib.optionalString stdenv.isLinux ''
     substituteInPlace modules/audio_device/linux/alsasymboltable_linux.cc \
       --replace-fail "libasound.so.2" "${alsa-lib}/lib/libasound.so.2"
   '';
@@ -73,46 +115,59 @@ stdenv.mkDerivation (finalAttrs: {
     echo "generate_location_tags = true" >> build/config/gclient_args.gni
   '';
 
-  gnFlags = [
-    # webrtc uses chromium's `src/build/BUILDCONFIG.gn`. many of these flags
-    # are copied from pkgs/applications/networking/browsers/chromium/common.nix.
-    ''target_os="linux"''
-    ''target_cpu="${chromiumRosettaStone.cpu stdenv.hostPlatform}"''
-    ''pkg_config="${stdenv.cc.targetPrefix}pkg-config"''
-    "use_sysroot=false"
-    "is_clang=false"
-    "treat_warnings_as_errors=false"
-    "use_llvm_libatomic=false"
-    "use_custom_libcxx=false"
+  gnFlags =
+    lib.optionals stdenv.isLinux [
+      # webrtc uses chromium's `src/build/BUILDCONFIG.gn`. many of these flags
+      # are copied from pkgs/applications/networking/browsers/chromium/common.nix.
+      ''target_os="linux"''
+      ''pkg_config="${stdenv.cc.targetPrefix}pkg-config"''
+      "use_sysroot=false"
+      "is_clang=false"
 
-    # https://github.com/signalapp/ringrtc/blob/main/bin/build-desktop
-    "rtc_build_examples=false"
-    "rtc_build_tools=false"
-    "rtc_use_x11=false"
-    "rtc_enable_sctp=false"
-    "rtc_libvpx_build_vp9=true"
-    "rtc_disable_metrics=true"
-    "rtc_disable_trace_events=true"
-    "is_debug=false"
-    "symbol_level=1"
-    "rtc_include_tests=false"
-    "rtc_enable_protobuf=false"
-
-    ''rust_sysroot_absolute="${buildPackages.rustc}"''
-
-    # Build using the system toolchain (for Linux distributions):
-    #
-    # What you would expect to be called "target_toolchain" is
-    # actually called either "default_toolchain" or "custom_toolchain",
-    # depending on which part of the codebase you are in; see:
-    # https://chromium.googlesource.com/chromium/src/build/+/3c4595444cc6d514600414e468db432e0f05b40f/config/BUILDCONFIG.gn#17
-    ''custom_toolchain="//build/toolchain/linux/unbundle:default"''
-    ''host_toolchain="//build/toolchain/linux/unbundle:default"''
-  ]
-  ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
-    ''host_toolchain="//build/toolchain/linux/unbundle:host"''
-    ''v8_snapshot_toolchain="//build/toolchain/linux/unbundle:host"''
-  ];
+      # Build using the system toolchain (for Linux distributions):
+      #
+      # What you would expect to be called "target_toolchain" is
+      # actually called either "default_toolchain" or "custom_toolchain",
+      # depending on which part of the codebase you are in; see:
+      # https://chromium.googlesource.com/chromium/src/build/+/3c4595444cc6d514600414e468db432e0f05b40f/config/BUILDCONFIG.gn#17
+      ''custom_toolchain="//build/toolchain/linux/unbundle:default"''
+      ''host_toolchain="//build/toolchain/linux/unbundle:default"''
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      ''target_os="mac"''
+      ''mac_deployment_target="${stdenv.hostPlatform.darwinMinVersion}"''
+      "use_sysroot=true"
+      "is_clang=true"
+      "use_lld=false"
+      "clang_use_chrome_plugins=false"
+      "clang_use_raw_ptr_plugin=false"
+      "use_clang_modules=false"
+      ''clang_base_path="${llvmCcAndBintools}"''
+      ''custom_toolchain="//build/toolchain/mac:clang_${chromiumRosettaStone.cpu stdenv.hostPlatform}"''
+    ]
+    ++ [
+      ''target_cpu="${chromiumRosettaStone.cpu stdenv.hostPlatform}"''
+      # https://github.com/signalapp/ringrtc/blob/main/bin/build-desktop
+      "rtc_build_examples=false"
+      "rtc_build_tools=false"
+      "rtc_use_x11=false"
+      "rtc_enable_sctp=false"
+      "rtc_libvpx_build_vp9=true"
+      "rtc_disable_metrics=true"
+      "rtc_disable_trace_events=true"
+      "is_debug=false"
+      "symbol_level=1"
+      "rtc_include_tests=false"
+      "rtc_enable_protobuf=false"
+      "treat_warnings_as_errors=false"
+      "use_llvm_libatomic=false"
+      "use_custom_libcxx=false"
+      ''rust_sysroot_absolute="${buildPackages.rustc}"''
+    ]
+    ++ lib.optionals (stdenv.isLinux && stdenv.buildPlatform != stdenv.hostPlatform) [
+      ''host_toolchain="//build/toolchain/linux/unbundle:host"''
+      ''v8_snapshot_toolchain="//build/toolchain/linux/unbundle:host"''
+    ];
   ninjaFlags = [ "webrtc" ];
 
   installPhase = ''
@@ -128,6 +183,6 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://github.com/SignalApp/webrtc";
     license = lib.licenses.bsd3;
     maintainers = [ ];
-    platforms = lib.platforms.linux;
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
   };
 })

